mutiny-core/src/lib.rs

```diff
@@ -124,10 +141,53 @@ impl<S: MutinyStorage> MutinyWallet<S> {
         Ok(())
     }
 
+    /// Starts a background process that will watch for nostr wallet connect events
+    pub async fn start_nostr_wallet_connect(&self, from_node: PublicKey) {
+        let nostr = self.nostr.clone();
+        let nm = self.node_manager.clone();
+        spawn_local(async move {
+            let client = Client::new(&nostr.primary_key);
+            client
+                .add_relays(nostr.relays.clone())
+                .await
+                .expect("Failed to add relays");
+            client.connect().await;
+            client.subscribe(vec![nostr.create_nwc_filter()]).await;
+
+            // broadcast NWC info event
+            // todo we only need to broadcast the first time
+            if let Ok(event) = nostr.create_nwc_info_event() {
+                if let Err(e) = client.send_event(event).await {
+                    log_warn!(nm.logger, "Error sending NWC info event: {e}");
+                }
+            }
+
+            // handle NWC requests
+            let mut notifications = client.notifications();
+            while let Ok(notification) = notifications.recv().await {
+                if let RelayPoolNotification::Event(_url, event) = notification {
+                    if event.kind == Kind::Custom(23194) {
+                        match nostr.handle_nwc_request(event, &nm, &from_node).await {
+                            Ok(Some(event)) => {
+                                if let Err(e) = client.send_event(event).await {
+                                    log_warn!(nm.logger, "Error sending NWC event: {e}");
+                                }
+                            }
+                            Ok(None) => {} // no response
+                            Err(e) => {
+                                log_error!(nm.logger, "Error handling NWC request: {e}");
+                            }
+                        }
+                    }
+                }
```

Comment:
Should this be wrapped in a loop? Otherwise if `notifications.recv()` throw an error (assuming connection error), then it'll stop listening to connections?